src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -74,8 +90,10 @@ public override FormulaValue Visit(TableNode node, SymbolContext context)
             var args = new FormulaValue[len];
             for (var i = 0; i < len; i++)
             {
+                CheckCancel();
```

Comment:
>CheckCancel [](http://example.com/codeflow?start=16&length=11)

We have a cooperative-abort model here, checking for CancellationToken in any loop. (We should have a Code Analyzer enforce this).  For safety, just do all loops - don't bother trying to rationalize between large vs. small loops.

Reply:
We can probably come up with "CancellableLoop" or "CancellableForeach" primitive for this purpose